src/GitVersion.LibGit2Sharp/Git/GitRepository.cs

```diff
@@ -53,10 +54,20 @@ public void Dispose()
 
         public ICommit FindMergeBase(ICommit commit, ICommit otherCommit)
         {
-            var mergeBase = repositoryInstance.ObjectDatabase.FindMergeBase((Commit)commit, (Commit)otherCommit);
-            return new Commit(mergeBase);
+            return new OperationWithExponentialBackoff<LockedFileException, ICommit>(new ThreadSleep(), log, () =>
```

Comment:
Looks good, do you think we can move the retry from the `GitRepository` and from the LibGit2Sharp implementation upper the stack to the consumer. That will make it more consistent.